api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -38,52 +59,316 @@ class Meta:
             "end_date",
         ]
         extra_kwargs = {
-            "user": {
-                "read_only": False,
-            },
+            "begin_date": {"required": True},
+            "end_date": {"required": True},
+            "begin_time": {"required": True},
+            "end_time": {"required": True},
+            "recurrence_in_days": {"required": True},
         }
 
     def validate(self, data: dict[str, Any]) -> dict[str, Any]:
-        begin_time = data.get("begin_time", getattr(self.instance, "begin_time", None))
-        end_time = data.get("end_time", getattr(self.instance, "end_time", None))
-
-        if end_time < begin_time:
-            raise ValidationErrorWithCode(
-                "Begin time cannot be after end time.",
-                ValidationErrorCodes.RESERVATION_BEGIN_AFTER_END,
-            )
-
-        begin_date = data.get("begin_date", getattr(self.instance, "begin_date", None))
-        end_date = data.get("end_date", getattr(self.instance, "end_date", None))
+        begin_date: datetime.date = self.get_or_default("begin_date", data)
+        end_date: datetime.date = self.get_or_default("end_date", data)
+        begin_time: datetime.time = self.get_or_default("begin_time", data)
+        end_time: datetime.time = self.get_or_default("end_time", data)
+        reservation_unit: ReservationUnit = self.get_or_default("reservation_unit", data)
 
         if end_date < begin_date:
-            raise ValidationErrorWithCode(
-                "Begin date cannot be after end date.",
-                ValidationErrorCodes.RESERVATION_BEGIN_AFTER_END,
-            )
+            msg = "Begin date cannot be after end date."
+            raise ValidationError(msg, code=error_codes.RESERVATION_BEGIN_DATE_AFTER_END_DATE)
 
-        user = self.context.get("request").user
-        if user.is_anonymous:
-            user = None
+        if end_time < begin_time:
```

Comment:
<= since can't allow zero-length reservations

Reply:
Fixed this and added tests.